### ISSUE TITLE
singleSelect not selecting

### ISSUE THREAD

**Issue by DeAustin** [paulopmx/Flexigrid#7]
Hello, I am using Flexigrid for jQuery -  v1.1 I noticed that while in singleSelect mode that no rows ever get selected.

I modified addRowProp line 691 from

if (p.singleSelect && ! g.multisel ) {
    $(this).siblings().removeClass('trSelected');
    $(this).toggleClass('trSelected');
}

to

if (p.singleSelect && ! g.multisel ) {
    $(this).siblings().removeClass('trSelected');
    $(this).addClass('trSelected');
}

this seemed to resolve the problem.

**Comment by paulopmx** [comment]
Hi DeAustin,

Please Fork.

Then request for a pull.

Thanks.

On Tuesday, May 15, 2012, DeAustin wrote:

> Hello, I am using Flexigrid for jQuery -  v1.1 I noticed that while in
> singleSelect mode that no rows ever get selected.
> 
> I modified addRowProp line 691 from
> 
> if (p.singleSelect && ! g.multisel ) {
>        $(this).siblings().removeClass('trSelected');
>        $(this).toggleClass('trSelected');
> }
> 
> to
> 
> if (p.singleSelect && ! g.multisel ) {
>        $(this).siblings().removeClass('trSelected');
>        $(this).addClass('trSelected');
> }
> 
> this seemed to resolve the problem.
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/paulopmx/Flexigrid/issues/7

## 

Paulo P. Marinas
Chief Technology Officer

**Comment by DeAustin** [comment]
The solution I provided did not fully resolve the problem, as it did not allow for the single items to be deselected. However this does:

if (!p.singleSelect) {
    $(this).toggleClass('trSelected');
} else if (p.singleSelect && ! g.multisel ) {
    $(this).siblings().removeClass('trSelected');
    $(this).toggleClass('trSelected');
}

The multiple select using shift and ctrl keys has some very strange behavior I noticed, this may need looking at by someone. For now I would remove my fork as it is incorrect.
